arbor/morph/segment_tree.cpp

```diff
@@ -12,6 +12,108 @@
 
 namespace arb {
 
+template<typename P>
+segment_tree fold(const segment_tree& tree,
+                  P predicate,
+                  std::pair<msize_t, msize_t> start,
+                  const segment_tree& acc={}) {
```

Comment:
I am a bit confused about the naming of the function arguments. If I understand correctly, `predicate` is actually a "sentinel" functor, i.e. the folding stops once it evaluates to true. Also I am not sure why the last argument is named `acc` - usually this would be called something like "initval". Anyways, would be useful to describe what this function is actually doing.

Reply:
Sure, I can add some more explanation to the function. It's an internal helper that appeared as a common pattern.

Reply:
Also, while it is kind of a fold, copy_if seems more to the point.

Reply:
Thanks for adding the documentation. I still think the description for the predicate is unclear, though. Otherwise I like it.